src/object/attributes/cache.rs

```diff
@@ -34,10 +34,13 @@ impl CacheAttributes {
     }
 
     /// Depth of the cache (e.g. L1, L2, ...)
+    ///
+    /// Note that following hardware nomenclature, cache depths normally start
+    /// at 1, corresponding to the L1 cache.
     #[doc(alias = "hwloc_cache_attr_s::depth")]
     #[doc(alias = "hwloc_obj_attr_u::hwloc_cache_attr_s::depth")]
-    pub fn depth(&self) -> NonZeroUsize {
-        NonZeroUsize::new(int::expect_usize(self.0.depth)).expect("Cache depths should start at 1")
+    pub fn depth(&self) -> usize {
+        int::expect_usize(self.0.depth)
```

Comment:
On second thought, we can't say for sure that no hardware would be crazy enough to expose an L0 cache. Some people call the x86 microcode cache an L0 cache...